src/query_plan/operation.rs

```diff
@@ -2776,6 +2873,44 @@ impl NormalizedFieldSelection {
     pub(crate) fn has_defer(&self) -> bool {
         self.field.has_defer() || self.selection_set.as_ref().is_some_and(|s| s.has_defer())
     }
+
+    fn can_add_to(
+        &self,
+        parent_type: &CompositeTypeDefinitionPosition,
+        schema: &ValidFederationSchema,
+    ) -> bool {
+        if &self.field.data().schema == schema {
+            match (parent_type, &self.field.data().field_position) {
```

Comment:
I believe you should be matching field parent against the parent (and not field type vs parent type) -> i.e. 

``` &self.field.data().schema == schema && &self.field.data().field_position.parent() == parent_type```